docs/unidling.md

```diff
@@ -0,0 +1,185 @@
+# Unidling Pods
+## Abstract
+
+Unidling is an event-driven feature that allows users to automatically create destination pods for
+an idled service when a connection is made to it.
+
+## Motivation
+
+Not all pods in a Kubernetes cluster will be active at all times.  When all of the pods
+resolving to a service are inactive, new ones must be created when a new request for that service
+is received.  
+
+An idled service is a service with zero endpoints.  The process of creating new pods for an idled
+service in response to a new request to that service is called "unidling".  This proposal covers
+how unidling should be performed in Kubernetes.
+
+Idling is the process of determining which pods to stop and stopping them. Idling will be dealt
+with in a separate proposal and is out of scope here.  
+
+## Constraints and Assumptions
+
+- Unidling should be orthogonal to autoscaling and idling
+- No specific idling mechanism or autoscaler is assumed
+- The case where a service is backed by pods which are not managed by a replication controller: 
```

Comment:
I don't see a need to ever support that. Even singleton pods should be managed by replication controllers.